server/lib/nl/detection/date.py

```diff
@@ -31,12 +31,16 @@
 LAST_YEAR = [r'(?:in|during|over)(?: the)? (?:last|past|previous) year']
 
 YEAR_MONTH_RE = [
-    r'(in|after|on|before|since|by|util|from|between) (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)(?:,)? (\d{4})',
-    r'(in|after|on|before|since|by|util|from|between) (\d{4})(?:,)? (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)',
+    r'(in|after|on|before|since|by|until|from|between) (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)(?:,)? (\d{4})',
+    r'(in|after|on|before|since|by|until|from|between) (\d{4})(?:,)? (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)',
 ]
 
 # List of date preps that indicate single date
 _SINGLE_DATE_PREPS = ['in', 'on']
+# List of date preps that indicate that the base date is a start date
+_START_DATE_PREPS = ['after', 'since', 'from']
+# List of date preps that indicate that the base date is an end date
+_END_DATE_PREPS = ['before', 'by', 'until']
```

Comment:
To confirm, should we handle <= vs. < ?

Reply:
ah, yes, that would make sense, will do in a follow up PR